.travis.yml

```diff
@@ -151,6 +151,34 @@ jobs:
         - EXTRA_CXXFLAGS="-DDEBUG"
       script: echo "Not running any tests for a debug build."
 
+    - stage: Test different OS/CXX/Flags
+      os: linux
+      cache: ccache
+      env:
+        - BUILD_SYSTEM=cmake
+      addons:
+        apt:
+          sources:
+            - ubuntu-toolchain-r-test
+          packages:
+            - g++-5
+      install:
+        - cmake -H. -Bbuild '-DCMAKE_BUILD_TYPE=Release' '-DCMAKE_CXX_FLAGS_RELEASE=-O3' '-DCMAKE_CXX_FLAGS=-Wall -Wpedantic -Werror' '-DCMAKE_CXX_COMPILER=g++-5'
```

Comment:
Do we compile with -O3 now?

Reply:
The default "Release" flags for clang++ and g++ are `-O3 -DNDEBUG`. We have to keep assertions enabled, which is why I've manually overridden the flags. I can switch to `-O2` too if you'd like.

Reply:
I'd say for a start the Release build ought to be using an as similar set of flags as the current build as possible. Is it possible to make this the default in the CMake configuration rather than requiring overrides on the command line?